doc/dependency-resolution.md

```diff
@@ -1,356 +1,129 @@
+content items.
 
-We should make it so that this query only follows edges in the dependency graph
-for groups which contain any of the fields that have changed on the content
-item. The branch linked to above demonstrates that it is possible to restrict
-edges followed by a link_type (which is another name for 'group'). The recursive
-behaviour of this query depends on the boolean configuration setting for that
-group and should be factored into the query.
+## Determining the links to show with a content item
 
-All of this behaviour should be completed out-of-band from the publishing
-application's request cycle with the Publishing API. It is proposed that we
-isolate all of this dependency resolution work into a separate Sidekiq worker
-that is responsible for determining the set of content items that need to be
-updated and for actually presenting and sending those updates downstream. When
-the Publishing API receives any interaction that can affect dependencies, it
-need only place this content item on a 'dirty' queue for processing by the
-dependency resolution worker.
+At the point of representing a content item to a Content Store we need to load
+the details of every content item that has an explicit or implicit link from
+this content item.
 
-If required, we can separate the calculation of dependencies from sending the
-updates downstream into separate workers/queues. The important thing is that
-both of those tasks happen out-of-band to the request cycle.
+Explicit links are links which have been set for a content item through a
+PatchLinkSet request. eg a request of:
 
-Note: Bidirectional dependencies is documented separately later on.
+@TODO example
 
-- Link expansion
+Would store a link to 123 of type:
 
-It is proposed that we move link expansion into the publishing api and send the
-fully-pressed content item downstream to the content store. For each of the link
-groups that appear on the content item, it will need to look up the rules for
-those groups from the schemas and expand those fields from the dependent content
-items.
+@TODO exmaple
 
-We also need to consider a new data structure to represent recursively expanded
-links. It is proposed that the structure remain almost identical with the key
-difference that the links for dependent content items are recursively expanded
-in the links data. For now, I think we should include the expanded links as a
-separate field in the content item. Perhaps this can be called "expanded_links",
-or similar and we can send it alongside the unexpanded links. As we migrate
-applications to use dependency resolution, we should make them use this field
-instead of "links" and look at retiring the expansion of links in content store.
+Thus any time either 456 or 123 changes a new representation of the content
+item is created for the Content Store.
 
-Here is a rough example of what this **might** look like:
+In addition to explicit links there are also implicit links between content
```

Comment:
As per our slack discussion, perhaps we could avoid implicit/explicit jargon altogether.

For example, instead of:

> In addition to explicit links there are also implicit links between content items.

You can explain:

> In addition to links that are set by publishing applications, content items items can be automatically linked by the publishing-api.